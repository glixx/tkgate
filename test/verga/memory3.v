module top;
   reg [7:0] m[0:255];
   reg [7:0] addr;
   reg x = 1'b0;
   integer i;
   wire [7:0] w;

   assign     w = m[3];

   always @(w)
     $display($time,": w changed to ",w);
   
   initial
     begin
	m[3][7:4] = 9;
	m[3][3:0] = 2;

	m[3] <= #2 7;
	m[3][7:4] <= #3 3;
	m[3] <= @(posedge x) 8'h99;

	$display($time,": m[3]=",m[3]," w=",w);
	for (i = 0;i < 20;i = i+1)
	  #1 $display($time,": m[3]=",m[3]," w=",w);
     end // initial begin

   initial
     begin
	#6 x = 1'b1;
	#3 x = 1'b0;
	m[3] = 8'h42;
	#3 x = 1'b1;
	#3 m[4] = 9;
     end
   
endmodule // top
