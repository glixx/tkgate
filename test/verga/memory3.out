0: m[3]=8'h92 w=8'hz
0: w changed to 8'hx
0: w changed to 8'h92
1: m[3]=8'h92 w=8'h92
2: m[3]=8'h7 w=8'h92
2: w changed to 8'h7
3: m[3]=8'h37 w=8'h7
3: w changed to 8'h37
4: m[3]=8'h37 w=8'h37
5: m[3]=8'h37 w=8'h37
6: m[3]=8'h37 w=8'h37
6: w changed to 8'h99
7: m[3]=8'h99 w=8'h99
8: m[3]=8'h99 w=8'h99
9: m[3]=8'h42 w=8'h99
9: w changed to 8'h42
10: m[3]=8'h42 w=8'h42
11: m[3]=8'h42 w=8'h42
12: m[3]=8'h42 w=8'h42
13: m[3]=8'h42 w=8'h42
14: m[3]=8'h42 w=8'h42
15: m[3]=8'h42 w=8'h42
16: m[3]=8'h42 w=8'h42
17: m[3]=8'h42 w=8'h42
18: m[3]=8'h42 w=8'h42
19: m[3]=8'h42 w=8'h42
20: m[3]=8'h42 w=8'h42
